src/components/Button/__tests__/Button.spec.jsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import {
+  expect,
+  test,
+} from '@playwright/experimental-ct-react';
+import {
+  mixPropTests,
+  propTests,
+} from '../../../../tests/playwright';
+import {
+  ButtonForTest,
+  ButtonForRefTest,
+} from './Button.story';
+
+test.describe('Button', () => {
+  test.describe('visual', () => {
+    [
+      ...propTests.afterLabelPropTest,
+      ...propTests.beforeLabelPropTest,
+      ...propTests.blockPropTest,
+      ...mixPropTests([
+        [
+          ...propTests.actionColorPropTest,
+          ...propTests.feedbackColorPropTest,
+          ...propTests.neutralColorPropTest,
+        ],
+        propTests.disabledPropTest,
+        propTests.priorityPropTest,
+      ]),
+      ...propTests.endCornerPropTest,
+      ...propTests.feedbackIconPropTest,
+      ...propTests.labelPropTest,
+      ...propTests.labelVisibilityPropTest,
+      ...propTests.sizePropTest,
+      ...propTests.startCornerPropTest,
+    ].forEach(({
+      name,
+      onBeforeTest,
+      props,
+    }) => {
+      test(name, async ({
+        mount,
+        page,
+      }) => {
+        if (onBeforeTest) {
+          await onBeforeTest(page);
+        }
+
+        const component = await mount(
+          <ButtonForTest
+            {...props}
+          />,
+        );
+
+        const screenshot = await component.screenshot();
+        expect(screenshot).toMatchSnapshot();
+      });
+    });
+  });
+
+  test.describe('non-visual', () => {
+    test('id', async ({ mount }) => {
+      const component = await mount(
+        <ButtonForTest
+          id="test-id"
+        />,
+      );
+
+      const id = await component.getAttribute('id');
+      expect(id).toBe('test-id');
+    });
+
+    test('ref', async ({ mount }) => {
+      const component = await mount(
```

Comment:
One of Playwright CT restrictions is that in can pass only primitives as prop. Otherwise story must be used. Check the documentation: https://playwright.dev/docs/test-components#test-stories

 This is reason why I came up with such weird test for ref.